sql/src/main/java/io/crate/execution/jobs/JobExecutionContext.java

```diff
@@ -207,14 +232,29 @@ public long kill() {
         return numKilled;
     }
 
-    private void finish() {
+    public void close() {
         if (failure != null) {
             finishedFuture.completeExceptionally(failure);
         } else {
             finishedFuture.complete(null);
         }
     }
 
+    private void finish() {
+        if (profiler.enabled()) {
+            if (LOGGER.isTraceEnabled()) {
+                LOGGER.trace("Profiling is enabled. JobExecutionContext will not be closed until results are collected!");
+                LOGGER.trace("Profiling results for job {}: {}", jobId, profiler.getAsMap());
+            }
+        } else {
+            close();
+        }
+    }
+
+    public ImmutableMap<String, Long> executionTimes() {
```

Comment:
Can we change the signature to return `Map` only? Leaking implementation details across boundaries is not nice. Especially if it's 3rd party classes.